File: Pattern38.java
import java.util.Scanner;

/**
 * *
 ** **
 * * * *
 * * * *
 * ** *
 * ** *
 * * * *
 * * * *
 ** **
 * *
 * 
 * 
 */
public class Pattern38 {

    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter the number of times you want to execute : ");
        int n = scanner.nextInt();
        scanner.close();
        Pattern(n);

    }

    // Pattern38 constructor
    public static void Pattern(int n) {
        n = n * 2;
        for (int row = 1; row <= n; row++) {

            for (int col = 1; col <= n; col++) {
                if (row == col || col + row == n + 1 || col == 1 || col == n)
                    System.out.print("*");
                else
                    System.out.print(" ");

            }
            System.out.println();

        }
    }
}

File: Pattern38.c
#include <stdio.h>

void main()
{

    int n;
    printf("Enter the row number: ");
    scanf("%d", &n);

    for (int i = 1; i <= n; i++)
    {

        for (int j = 1; j <= i; j++)
        {
            if (j == i || j == 1)
            {
                printf("*");
            }
            else
            {
                printf(" ");
            }
        }

        for (int space = 2 * (n - i); space > 0; space--)
        {
            printf(" ");
        }

        for (int k = 1; k <= i; k++)
        {
            if (k == 1 || k == i)
            {
                printf("*");
            }
            else
            {
                printf(" ");
            }
        }
        printf("\n");
    }

    for (int i = n; i > 0; i--)
    {

        for (int j = 1; j <= i; j++)
        {
            if (j == 1 || j == i)
            {
                printf("*");
            }
            else
            {
                printf(" ");
            }
        }

        int spaces = 2 * (n - i);
        for (int j = 1; j <= spaces; j++)
        {
            printf(" ");
        }

        for (int j = 1; j <= i; j++)
        {
            if (j == 1 || j == i)
                printf("*");
            else
                printf(" ");
        }
        printf("\n");
    }
}